### PR TITLE
[webpack-config-terra] Upgrade to sass (dart-sass)

### LINKED ISSUE #676: [webpack-config-terra] Replace node-sass with sass (dart-sass)
# Feature Request

## Description

As of October 2020 [node-sass]9https://www.npmjs.com/package/node-sass) is [deprecated](https://sass-lang.com/blog/libsass-is-deprecated). The [sass](https://www.npmjs.com/package/sass) npm package is the replacement. 

While the upgrade to this package is incredibly easy ([see this branch](https://github.com/cerner/terra-toolkit/compare/main...switch-to-dart-sass)), this change has a ripple effect on screenshot comparisons downstream.

Specifically the sass package calculates a greater precision than the node-sass package.

For example, the sass package has a precision of 10 which is [non adjustable](https://github.com/sass/dart-sass#javascript-api). See [line-height](https://github.com/cerner/terra-core/blob/main/packages/terra-base/src/Base.scss#L26) in terra-base.
```css
body {
    color: #1c1f21;
    color: var(--terra-base-color, #1c1f21);
    font-size: 1rem;
    height: 100%;
    height: var(--terra-base-body-height, 100%);
    line-height: 1.4285714286;
    line-height: var(--terra-base-line-height, 1.4285714286)
}
```
Where as the node-sass package has a default precision of [5](https://github.com/sass/node-sass#precision)
```css
body {
    color: #1c1f21;
    color: var(--terra-base-color, #1c1f21);
    font-size: 1rem;
    height: 100%;
    height: var(--terra-base-body-height, 100%);
    line-height: 1.42857;
    line-height: var(--terra-base-line-height, 1.42857)
}
```
The effect in this specific instance on wdio tests is that content down the page increasingly gets 'off' according to the screenshot comparison:
open_fullscreen_modal

## Options

1. Consider this passive and require consuming teams and ourselves to update screenshots.
2. Consider this passive, hunt down any css variable with greater than 5 precision and update it prior to release.
3. Consider this non passive and include it in the next mvb, or release an mvb.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This PR updates webpack-config-terra from node-sass to sass (dart-sass).

Closes #676 

### Additional Details

@cerner/terra

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/UpgradeGuide.y.tool.mdx [hunk 1]
```diff
@@ -4,6 +4,16 @@ import { Badge } from '@cerner/webpack-config-terra/package.json?dev-site-packag
 
 # webpack-config-terra Upgrade Guide
 
+## Changes from @cerner/webpack-config-terra 1.0.0 to @cerner/webpack-config-terra 2.0.0
```
Comment (mjhenkes) [inline packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/UpgradeGuide.y.tool.mdx]: Let's make this it's own document.

Comment (pranav300) [reply packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/UpgradeGuide.y.tool.mdx]: Updated here https://github.com/cerner/terra-toolkit/pull/685/commits/374e8ac4d1d266c71a19a2f6598f1e8ef31a61f4.

**Comment by mjhenkes** [comment]
Can you reverse the order of the upgrade guides? version 2 above version 1

File: packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/upgrade-guide.6/version-2-upgrade-guide.2.tool.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## Changes from @cerner/webpack-config-terra 1.0.0 to @cerner/webpack-config-terra 2.0.0
+
+#### Removed node-sass
+
+We have removed `node-sass` from dependencies and switched to `sass (dart-sass)`.
+
+#### Update WDIO snapshots
+
+The previously used `node-sass` had a precision of 5, whereas `sass` has an unadjustable precision of 10. Hence it is required for all the users upgrading to @cerner/webpack-config-terra 2.0.0 to update their WDIO snapshots.
```
Comment (mjhenkes) [inline packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/upgrade-guide.6/version-2-upgrade-guide.2.tool.mdx]:
```suggestion
The previously used `node-sass` had a precision of 5, whereas `sass` has an unadjustable precision of 10. Hence consumers may need to regenerate screenshots to account for slightly larger css values.
```

File: packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/upgrade-guide.6/version-2-upgrade-guide.2.tool.mdx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+<Badge />
+
+# webpack-config-terra Upgrade Guide
+
+## Changes from @cerner/webpack-config-terra 1.0.0 to @cerner/webpack-config-terra 2.0.0
+
+#### Removed node-sass
+
+We have removed `node-sass` from dependencies and switched to `sass (dart-sass)`.
```
Comment (mjhenkes) [inline packages/terra-toolkit-docs/src/terra-dev-site/tool/webpackConfigTerra/upgrade-guide.6/version-2-upgrade-guide.2.tool.mdx]:
```suggestion
We have removed [node-sass](https://github.com/sass/node-sass) from dependencies and switched to [sass (dart-sass)](https://github.com/sass/dart-sass).
```

File: packages/webpack-config-terra/package.json [hunk 4]
```diff
@@ -48,11 +48,11 @@
     "glob": "^7.1.1",
     "memory-fs": "^0.4.1",
     "mini-css-extract-plugin": "^1.3.1",
-    "node-sass": "^5.0.0",
     "postcss-assets-webpack-plugin": "^4.0.0",
     "postcss-custom-properties": "^10.0.0",
```
Comment (mjhenkes) [inline packages/webpack-config-terra/package.json]: while we're here, I think updating to the new version of postcss-custom-properites is passive.
```suggestion
    "postcss-custom-properties": "^11.0.0",
```

**Review by mjhenkes — APPROVED** [review]

**Review by benbcai — APPROVED** [review]

**Pull request merged**